appmap/_implementation/configuration.py

```diff
@@ -72,21 +129,46 @@ def name(self):
     def packages(self):
         return self._config['packages']
 
+    @property
+    @lru_cache(maxsize=None)
+    def default(self):
+        root_dir = Env.current.root_dir
+        return {
+            'name': default_app_name(root_dir),
+            'packages': [{'path': p} for p in find_top_packages(root_dir)],
+        }
+
     @property
     @lru_cache(maxsize=None)
     def _config(self):
-        path = Path(Env.current.get("APPMAP_CONFIG", "appmap.yml")).resolve()
+        # Only use a default config if the user hasn't specified a
+        # config.
+        env_config = Env.current.get("APPMAP_CONFIG")
```

Comment:
Do we need to support `APPMAP_CONFIG` env var for code extensions? Feels like the initial flow would not ask user to set env vars.

Reply:
Ruby CLI supports passing `-c,` `--config` params to set custom path though. I think we could focus on using the default path in the first phase.

Reply:
`APPMAP_CONFIG` only needs to be used at runtime, the CLI doesn't need to support it (as Kevin said).

We can add support for `--config` to `appmap-agent-init` when the code extensions need it.